compiler/modules/front.syntax/parseExpr.bal

```diff
@@ -99,18 +99,33 @@ function parseRelationalExpr(Tokenizer tok) returns Expr|err:Syntax {
         return bin;
     }
     else if t == "is" {
-        tok.setMode(MODE_TYPE_DESC);
-        check tok.advance();
-        InlineTypeDesc td = check parseInlineTypeDesc(tok);
-        tok.setMode(MODE_NORMAL);
-        TypeTestExpr typeTest = { td, left: expr, semType: resolveInlineTypeDesc(td) };
+        InlineTypeDesc td = check parseTestTypeDesc(tok);
+        TypeTestExpr typeTest = { td, left: expr, semType: resolveInlineTypeDesc(td), negated: false };
         return typeTest;
     }
+    if t == "!" {
+        check tok.advance();
+        Token? t2 = tok.current();
+        if t2 is "is" {
+            InlineTypeDesc td = check parseTestTypeDesc(tok);
+            TypeTestExpr typeTest = { td, left: expr, semType: resolveInlineTypeDesc(td), negated: true};
+            return typeTest;
+        }
+        return err:syntax("invalid operator");
+    }
     else {
         return expr;
     }
 }
 
+function parseTestTypeDesc(Tokenizer tok) returns InlineTypeDesc|err:Syntax {
```

Comment:
A better way to do this is

- call the function finishTypeTestExpr
- pass in `negated` as a parameter
- return a TypeTestExpr

Reply:
done